terraform/policies.tf

```diff
@@ -83,7 +94,7 @@ data "aws_iam_policy_document" "self_manage_iam_user" {
     ]
 
     resources = [
-      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/$${aws:username}",
+      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/*",
```

Comment:
Wont this allow them to deactivate other peoples MFA devices?

Reply:
The [DeactivateMFADevice](https://docs.aws.amazon.com/IAM/latest/APIReference/API_DeactivateMFADevice.html) API call takes both a username and a device ARN so I've assumed that the presence of both the user and device resources will limit it to deactivating devices from their own users.

We definitely need to test that before merging, though.

Reply:
It doesn't I tried it a minute ago :( I've just had a play around with this stuff in the pay dev account and the results are not pleasing.

If you have permission to deactivate you can deactivate any users MFA (and also delete if you had the delete permission).

I don't think the condition keys cover it either, since the user condition is testing the user making the request, not the target user of the request.

Reply:
@jfharden I've just tested this myself in a test account and I got different results to you. It looks like only the `:user/${aws:username}` resource is relevant to the `EnableMFADevice` and `DeactivateMFADevice` API calls, and the presence of the `:mfa/*` resource makes no difference. I get:

`An error occurred (AccessDenied) when calling the DeactivateMFADevice operation: User: arn:aws:iam::xxxxxxxxxxxx:user/test1 is not authorized to perform: iam:DeactivateMFADevice on resource: user test2 because no identity-based policy allows the iam:DeactivateMFADevice action`

So I don't think this allows users to deactivate other users' MFA devices because there's no policy explicitly allowing the action on anything other than `:user/${aws:username}`, which is the calling user.

This does mean we can clean the policy up a bit though. I'll look at doing that.

Reply:
Oh yeah you're right, the IAM policy actions definition https://docs.aws.amazon.com/service-authorization/latest/reference/list_identityandaccessmanagement.html for DeactivateMFADevice says the the user is the only resource that works in the resource limitation, so I think you're right the mfa bit can be totally removed from this statement.

That's good since it means a user can only deactivate their own device with the user resource on the line below.

Reply:
Yeah enable, resync, and deactivate are all targetted by user, so I think you can tidy this quite a bit and move all 3 actions into 1 statement with only the user resource.